util/dvsim/SimCfg.py

```diff
@@ -236,6 +196,46 @@ def __post_init__(self):
         # Run some post init checks
         super().__post_init__()
 
+
+    def _resolve_wave_format(self):
+        '''Decide on the correct wave format if waves are enabled.
+
+        This is called after reading the config file. If waves is None,
+        then simply exit, since there is nothing to do here. The tool which
+        must be set at this point, supports a limited list of formats. If waves
+        is set to 'default', then pick the first item on the list. If not,
+        waves is set to the desired format. In that case we need to check if
+        the tool actually supports it.
+        '''
+        if not self.waves or self.is_primary_cfg:
+            self.waves = 'none'
+            return
+
+        assert self.tool is not None
+
+        # If the user hasn't specified a dumping format (No argument supplied
+        # to --waves), we need to decide on a format for them. The supported
+        # list of wave formats is set in the tool's HJson configuration using
+        # the `supported_wave_formats` key. If that list is not set, we use
+        # 'vpd' by default and hope for the best. It that list if set, then we
+        # pick the first available format for which the waveform viewer exists.
+        if self.waves == 'default':
+            if self.supported_wave_formats:
+                self.waves = pick_wave_format(self.supported_wave_formats)
+            else:
+                self.waves = 'vpd'
+            return
+
+        # If the user has specified their preferred dumping format, use it. As
+        # a sanity check, error out if the chosen tool doesn't support the
+        # format, but only if we know about the tool. If not, we'll just assume
+        # they know what they're doing.
+        if self.supported_wave_formats and \
+           self.waves not in self.supported_wave_formats:
+            log.error('Chosen tool ({}) does not support wave dumping format '
+                      '{!r}.'.format(self.tool, self.waves))
```

Comment:
I think you've expunged the word "dump" from the code elsewhere. Assuming that was intentional, you probably want to drop it here too.

Reply:
I just wanted to drop the extra `--dump` switch. Updated this a bit anyway.